### PR TITLE
Vault image hardening

### PR DESCRIPTION
**Pull request by dreamer-nitj**
Changes to harden the vault image

Test Plan:
- Ran manual test to verify the stub/coverage

### CODE REVIEW

File: community_images/vault/hashicorp/coverage.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    kubectl exec -n "${NAMESPACE}" "${VAULT_CONTAINER}" -- vault write auth/kubernetes/config kubernetes_host="https://${KUBERNETES_PORT_443_TCP_ADDR}:${K8S_API_PORT}"
+    kubectl cp "${SCRIPTPATH}"/policy.hcl -n "${NAMESPACE}" "${VAULT_CONTAINER}":/tmp/
+    kubectl exec -n "${NAMESPACE}" "${VAULT_CONTAINER}" -- vault policy write webapp /tmp/policy.hcl
+    kubectl exec -n "${NAMESPACE}" "${VAULT_CONTAINER}" -- vault write auth/kubernetes/role/webapp \
+        bound_service_account_names=vault \
+        bound_service_account_namespaces=default \
+        policies=webapp \
+        ttl=24h
+
+    # # create the service account for webapp
```
Comment (dreamer-nitj) [inline community_images/vault/hashicorp/coverage.sh:52]: need to fix this part. this is just to verify that vault secret is accessible as per the policy

File: community_images/vault/hashicorp/.rfignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+opt/bitnami/common/licenses
+opt/bitnami/<change_me>/licenses
```
Comment (codervinod) [inline community_images/vault/hashicorp/.rfignore:2]: Please fix this file

File: community_images/vault/hashicorp/coverage.sh (hunk 1 above)
Comment (codervinod) [reply community_images/vault/hashicorp/coverage.sh:52]: Please fix this for merge

**Review by codervinod — COMMENTED** [review]
Vault used to have a UI. Dont they have it anymore or is it a separate image?

File: community_images/vault/hashicorp/image.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+input_registry:
+  registry: docker.io
+  account: library
+repo_sets:
+  - vault:
+      input_base_tag: "1.12.0"
+runtimes:
+  - type: k8s
+    script: k8s_coverage.sh
+    helm_repo: hashicorp/vault
```
Comment (codervinod) [inline community_images/vault/hashicorp/image.yml:32]: Please add helm: repo, url, chart as per new format.

Comment (dreamer-nitj) [reply community_images/vault/hashicorp/image.yml:33]: sorry, I think you mentioned about the new format in weekly meeting but i can't recall the details. i don't see anything on the page https://github.com/rapidfort/community-images/blob/main/image_yml.md either. can you pls remind me again about the new format?

Comment (codervinod) [reply community_images/vault/hashicorp/image.yml:33]: helm.repo | helm repo to use for k8s runtime, ex: 'bitnami' | "" | yes
-- | -- | -- | --
helm.repo_url | helm repo URL to use for k8s runtime, ex: 'https://charts.bitnami.com/bitnami' | "" | yes
helm.chart | helm chart to use for k8s runtime, ex: 'nats' | "" | yes

**Pull request merged**